features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/voicemessages/timeline/VoiceMessageCache.kt

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2023 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.features.messages.impl.voicemessages.timeline
+
+import com.squareup.anvil.annotations.ContributesBinding
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import io.element.android.libraries.di.AppScope
+import io.element.android.libraries.di.CacheDirectory
+import java.io.File
+
+/**
+ * Manages the local disk cache for a voice message.
+ */
+interface VoiceMessageCache {
+
+    /**
+     * Factory for [VoiceMessageCache].
+     */
+    fun interface Factory {
+        /**
+         * Creates a [VoiceMessageCache] for the given Matrix Content (mxc://) URI.
+         *
+         * @param mxcUri the Matrix Content (mxc://) URI of the voice message.
+         */
+        fun create(mxcUri: String): VoiceMessageCache
+    }
+
+    /**
+     * The file path of the voice message in the cache directory.
+     * NB: This doesn't necessarily mean that the file exists.
+     *
+     * @return the file path of the voice message in the cache directory.
+     */
+    val cachePath: String
```

Comment:
That's a file path (as in full path to file e.g. `/a/b/c/d/something.ogg`) not a directory.
I get it was not clear enough from the docstring, is there a better way to explain it in English?

Reply:
I see - I was a bit confused when I first saw the interface what it's for. I thought this is used to provide access to the file. But I think it's used to create the mediaUri [here](https://github.com/vector-im/element-x-android/pull/1503/files/b23f1d48f553ecaf51c0c6a37a14e6b842eb3aab#diff-ead1cbb0afc8a632d8782728759c8db7c31fd5abbc3412d7407c127aac4e7d06R132)?

It's not a big problem if we need it in this format - happy to leave as is!

Reply:
Both.
The idea is that we map one mxc:// content uri to one file in the cache via a simple mapping: `"${cacheDir.path}/$CACHE_VOICE_SUBDIR/${mxcUri2FilePath(mxcUri)}"` .

So once we have a content uri we can initialize an instance of this class and:
1) it will give us the path to the cache file using `cachePath`
2) it will tell us if the file at such path actually exists with `isInCache()` (i.e. cache hit) or not (i.e. cache miss)
3) it will give use the option to fill the cache with such content (`moveToCache()`)

Perhaps we can find better method names?